app/src/main/java/app/grapheneos/pdfviewer/fragment/PasswordPromptFragment.kt

```diff
@@ -0,0 +1,78 @@
+package app.grapheneos.pdfviewer.fragment
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextUtils
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.WindowManager
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import app.grapheneos.pdfviewer.PdfViewer
+import app.grapheneos.pdfviewer.R
+import app.grapheneos.pdfviewer.databinding.PasswordDialogFragmentBinding
+import com.google.android.material.textfield.TextInputEditText
+
+class PasswordPromptFragment : DialogFragment() {
+
+    private lateinit var passwordDialogFragmentBinding: PasswordDialogFragmentBinding
+    private lateinit var passwordEditText : TextInputEditText
+
+    private fun enableDisablePositiveButton(enable: Boolean) {
+        val positiveButton: Button =
+            (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+        if (enable) {
```

Comment:
Should be simplified to `positiveButton.isEnabled = enable` (framework checks if `isEnabled == enabled` itself),
and `enable` shouldn't be passed as an argument to simplify code in `onTextChanged()` and `onStart()`